ntoskrnl/ex/time.c

```diff
@@ -30,6 +30,108 @@ ULONG ExpTimerResolutionCount = 0;
 
 /* FUNCTIONS ****************************************************************/
 
+/*++
+ * If successful, this function sets the following global variables:
+ * ExpTimeZoneInfo
+ * ExpTimeZoneBias
+ *--*/
+static
+BOOLEAN
+ExpGetTimeZoneId(
```

Comment:
The function is called Exp**Get**TimeZoneId, which implies that it doesn't modify any global state, but it does. And it even does on failure. I suggest instead of modifying ExpTimeZoneBias in this function, to just return it in a 3rd parameter and let the caller set the global variable on success.

Reply:
Added third parameter to return NewTimeZoneBias. Thanks.